core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -797,6 +819,43 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
     }
   }
 
+  /**
+   * Check and delete specified event log according to the max log age defined by the user.
+   */
+  private def checkAndCleanLog(logPath: String): Unit = Utils.tryLog {
+    val maxTime = clock.getTimeMillis() - conf.get(MAX_LOG_AGE_S) * 1000
+    val expiredLog = Try {
+      val log = listing.read(classOf[LogInfo], logPath)
+      if (log.lastProcessed < maxTime) Some(log) else None
+    } match {
+      case Success(log) => log
+      case Failure(_: NoSuchElementException) => None
+      case Failure(e) => throw e
+    }
+
+    expiredLog.foreach { log =>
+      log.appId.foreach { appId =>
+        listing.view(classOf[ApplicationInfoWrapper])
```

Comment:
I'm not following. Why do you need to iterate and filter by appId instead of just fetching the entry directly? The appId is the key.

Reply:
Thanks a lot.